Added "add product", catalog and etc

Gradder opened the pull request.

Gradder committed:
Added "add product", catalog and etc

vakhonin commented on src/main/java/com/epam/onlineshop/entities/Product.java:
```diff
@@ -23,4 +23,7 @@
     private Integer price;
+
+    @Column(name = "imageurl")
+    private String imageLink;
```
Так так так... что то мне не нравится эта тема. Давайте посмотрим в сторону BLOB ?

anveslav replied:
Зачем, здесь просто текст, я думаю он даже 255 символов не будет превышать

vakhonin replied:
Мы хотим каждый раз подтягивать маленькую картинку продукты из физической памяти ??? Может быть картинку засунем сразу в поле ? поэтому BLOB

anveslav replied:
А данные в базе не в физической памяти хранятся ?

vakhonin replied:
У нас H2 настроена сейчас так, что она в оперативке вся держится )

lostforstupid commented on src/main/resources/data.sql:
```diff
@@ -1,6 +1,7 @@
-INSERT INTO user VALUES (1, 'Address', false, 'pass', 'ADMIN', 'admin');
+INSERT INTO user (id, role, username, password, is_Blocked, address) VALUES
+    ('1', 'ADMIN', 'admin', 'pass', 'false', 'Address');
```
А как id снова оказался в кавычках?)

lostforstupid replied:
Кстати, никого ещё не напрягает 1й юзер, а не 0й?)

anveslav replied:
Почему 0? Кстати, у нас в базе генерация айдишников не автоматическая?

vakhonin replied:
да, айдишники в базе нумеруются с 1.

Gradder replied:
Что мне скинули то и залил -_-

vakhonin replied:
Опа. Почему у нас так происходит ?

Gradder replied:
Нужно было для демонстрации

vakhonin replied:
Ок

anveslav commented on src/main/java/com/epam/onlineshop/controllers/ProductController.java:
```diff
@@ -23,16 +28,36 @@ public ModelAndView showProducts() {
-    public ModelAndView addProduct(@ModelAttribute("product") Product product) {
+    public ModelAndView addProduct(@ModelAttribute("product") Product product, @RequestParam("file") MultipartFile file) {
         ModelAndView catalog = new ModelAndView();
+        RandomString randomString = new RandomString(6);
```
Почему 6?

Gradder replied:
Это временно, я тестил и не доделал

anveslav commented on src/main/java/com/epam/onlineshop/controllers/ProductController.java:
```diff
@@ -23,16 +28,36 @@ public ModelAndView showProducts() {
+                Path currentRelativePath = Paths.get("");
+                String s = currentRelativePath.toAbsolutePath().toString();
+                BufferedOutputStream stream =
+                        new BufferedOutputStream(new FileOutputStream(new File(s + "/src/main/resources/static/images/products/" + name + ".jpg")));
```
Очень много кода в методе, можно куда-нибудь вынести часть логики?

Gradder replied:
Думаю вынесу куда нибудь в utils

anveslav commented on src/main/java/com/epam/onlineshop/entities/Product.java:
```diff
@@ -23,4 +23,7 @@
     @Column(name = "price", nullable = false)
     private Integer price;
+
+    @Column(name = "imageurl")
```
лучше image_url

vakhonin replied:
да, я скидывал некоторое соглашение об именах

anveslav commented on src/main/java/com/epam/onlineshop/utils/RandomString.java:
```diff
@@ -0,0 +1,62 @@
+        this(21);
+    }
+
+}
```
Что-то непонятно с порядком методов, полей и конструкторов в классе

Gradder replied:
Тоже в процессе работы, доделаю чуть позже

Gradder committed:
- Fixed file upload limitation
- New class for upload
- Fixed all troubles with images
- Fixed name generating for images
- Changed name for image link field in database

Gradder committed:
- Fixed file upload limitation
- New class for upload
- Fixed all troubles with images
- Fixed name generating for images
- Changed name for image link field in database

Gradder committed:
Merge branch 'master' into addProductFormAndCatalog

# Conflicts:
#	src/main/java/com/epam/onlineshop/entities/Product.java
#	src/main/resources/META-INF/resources/WEB-INF/jsp/cartList.jsp
#	src/main/resources/META-INF/resources/WEB-INF/jsp/header.jsp
#	src/main/resources/application.properties
#	src/main/resources/data.sql

Gradder committed:
Added temporary count for products
